dmt/A_mappers/ros_A_mapper.py

```diff
@@ -0,0 +1,1024 @@
+            f.write('      if(%s_lengthT > %s_length)\n' % (self.name, self.name))
+            f.write('        this->%s = '
+                    '(%s*)realloc(this->%s, %s_lengthT * sizeof(%s));\n' %
+                    (self.name, self.type, self.name, self.name, self.type))
+            f.write('      %s_length = %s_lengthT;\n' % (self.name, self.name))
+            # copy to array
+            f.write('      for( uint32_t i = 0; i < %s_length; i++){\n' %
+                    (self.name))
+            c.deserialize(f)
+            f.write('        memcpy( &(this->%s[i]), &(this->st_%s), sizeof(%s));\n'
+                    % (self.name, self.name, self.type))
+            f.write('      }\n')
+        else:
+            c = self.cls(self.name+"[i]", self.type, self.bytes)
+            f.write('      for( uint32_t i = 0; i < %d; i++){\n' % (self.size))
+            c.deserialize(f)
+            f.write('      }\n')
+
+    def convert_from(self, f, spacing):
+        f.write(spacing + '{\n')
+        if self.size is None:
+            # Use length
+            f.write(spacing + '  uint32_t length = var->%s.nCount;\n' % self.name)
+            f.write(spacing + '  %s_length = length;\n' % self.name)
+        else:
+            # Copy all
+            f.write(spacing + '  uint32_t length = %d;\n' % self.size)
+        f.write(spacing + '  memcpy(%s, var->%s, length * sizeof(*%s));\n' %
+                (self.name, self.name, self.name))
+        f.write(spacing + '}\n')
+
+
+    def convert_to(self, f, spacing):
+        f.write(spacing + '{\n')
+        if self.size is None:
+            # Use length
+            f.write(spacing + '  uint32_t length = %s_length;\n' % self.name)
+            f.write(spacing + '  var->%s.nCount = length;\n' % self.name)
+        else:
+            # Copy all
+            f.write(spacing + '  uint32_t length = %d;\n' % self.size)
+        f.write(spacing + '  memcpy(var->%s, %s, length * sizeof(*%s));\n' %
+                (self.name, self.name, self.name))
+        f.write(spacing + '}\n')
+
+
+
+# Messages
+
+class Message:
+    """ Parses message definitions into something we can export. """
+    global ROS_TO_EMBEDDED_TYPES
+
+    def __init__(self, name: str, package: str, definition: str, md5: str) -> None:
+
+        self.name = name            # name of message/class
+        self.package = package      # package we reside in
+        self.md5 = md5              # checksum
+        self.includes = list()      # other files we must include
+
+        self.data = list()          # data types for code generation
+        self.enums = list()
+
+        self.message_text = definition
+
+        # parse definition
+        definition = definition.splitlines()
+        for line in definition:
+            # prep work
+            line = line.strip().rstrip()
+            value = None
+            # Remove the comments
+            if line.find("#") > -1:
+                line = line[0:line.find("#")]
```

Comment:
Should this be moved before the strip and rstrip calls?

Reply:
could be